### PR TITLE
Dont crash if credential not found

### PR DESCRIPTION
**Pull request by pmiossec**
GetCredentials() now don't crash if credentials are not found

It is a nominal case to read not existing credentials,
so we should not throw and just return null

(Based on PR #22 , so should be merged after)

### CODE REVIEW

File: AdysTech.CredentialManager/CredentialManager.cs [hunk 1]
```diff
@@ -20,16 +20,16 @@ public enum CredentialType : uint
 
         private static bool PromptForCredentials(string target, NativeCode.CredentialUIInfo credUI, ref bool save, out string user, out string password, out string domain)
         {
-            user = String.Empty;
-            password = String.Empty;
-            domain = String.Empty;
+            user = string.Empty;
```
Comment (mvadu) [inline AdysTech.CredentialManager/CredentialManager.cs]: I am not okay with this change. This does not add any value, and makes the code harder to read. Given `string` is an alias for `System.String`,  I follow the basic concept, if its a `string` variable, I will use `string` whereas if I am using any static methods I prefer to use `String`. Hope you agree.

File: AdysTech.CredentialManager/CredentialManager.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ private static bool PromptForCredentials(string target, NativeCode.CredentialUII
                     user = usernameBuf.ToString();
                     password = passwordBuf.ToString();
                     domain = domainBuf.ToString();
-                    if (String.IsNullOrWhiteSpace(domain))
+                    if (string.IsNullOrWhiteSpace(domain))
```
Comment (mvadu) [inline AdysTech.CredentialManager/CredentialManager.cs]: as mentioned about, this I prefer to keep as `String.IsNullOrWhiteSpace`

**Comment by mvadu** [comment]
Could you please squash the three commits into one?

File: AdysTech.CredentialManager/CredentialManager.cs (hunk 1 above)
Comment (pmiossec) [reply AdysTech.CredentialManager/CredentialManager.cs]: This is a strange convention. The normal convention (encouraged by Resharper, Stylecop, Roslyn and perhaps even the last version of Visual Studio) is to use only the built-in C# types everywhere and not use the corresponding .Net framework type.

>This does not add any value, and makes the code harder to read.

It's just a matter of habits. I feel the other way ;)

But anyway, you're the maintainer, so, done.

**Comment by pmiossec** [comment]
>Could you please squash the three commits into one?

Not sure it's a good idea. You will mix 2 feature commits with a big refactoring one in a unique commit...
IMHO, that's not a good idea to get a good history...

I let you squash if you still want to do it as you could easily do it yourself with github when you will merge the pull request. That's one of the 3 "merge" options provided by the "merge button".

**Comment by mvadu** [comment]
One more thing I missed in last comment. Could you please version info with a minor version change? Otherwise I can't push this as a new Nuget package.

**Comment by pmiossec** [comment]
done.

**Comment by mvadu** [comment]
Thank you for spending effort on this one @pmiossec I have published new version in Nuget.

**Pull request merged**